app/notifications/receive_notifications.py

```diff
@@ -1,17 +1,64 @@
-from flask import Blueprint
-from flask import current_app
-from flask import request
+from urllib.parse import unquote
 
-from app.errors import register_errors
+from flask import Blueprint, current_app, request
+from notifications_utils.recipients import normalise_phone_number
+
+from app.dao.services_dao import dao_fetch_services_by_sms_sender
+from app.dao.inbound_sms_dao import dao_create_inbound_sms
+from app.models import InboundSms
+from app.errors import (
+    register_errors,
+    InvalidRequest
+)
 
 receive_notifications_blueprint = Blueprint('receive_notifications', __name__)
 register_errors(receive_notifications_blueprint)
 
 
 @receive_notifications_blueprint.route('/notifications/sms/receive/mmg', methods=['POST'])
 def receive_mmg_sms():
+    """
+    {
+        'MSISDN': '447123456789'
+        'Number': '40604',
+        'Message': 'some+uri+encoded+message%3A',
+        'ID': 'SOME-MMG-SPECIFIC-ID',
+        'DateRecieved': '2017-05-21+11%3A56%3A11'
+    }
+    """
     post_data = request.get_json()
-    post_data.pop('MSISDN', None)
-    current_app.logger.info("Recieve notification form data: {}".format(post_data))
+    potential_services = dao_fetch_services_by_sms_sender(post_data['Number'])
+
+    if len(potential_services) != 1:
+        current_app.logger.error('')
+        raise InvalidRequest(
+            'Inbound number "{}" not associated with exactly one service'.format(post_data['Number']),
+            status_code=400
+        )
+
+    service = potential_services[0]
+
+    inbound = create_inbound_sms_object(service, post_data)
+
+    current_app.logger.info('{} received inbound SMS with reference {}'.format(service.id, inbound.provider_reference))
+
+    return 'RECEIVED', 200
```

Comment:
When this endpoint was created in https://github.com/alphagov/notifications-api/commit/77f520acba154bddc1903169171e6157faf9b851, becca said "returns the requested RECEIVED", so i wasn't sure if changing this would impact MMG in any way